docs/DeveloperGuide.md

```diff
@@ -3,36 +3,144 @@
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+- [CS2113 Course Website](https://nus-cs2113-ay2324s2.github.io/website/index.html)
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
-
+The financial manager application is designed using an object-oriented approach, focusing on user authentication, transaction management, and user interaction. The core components include:
+
+- **Main Application Loop**: Handles the initialization of the application, including loading data from storage, authenticating the user, and processing user commands until the application exits.
```

Comment:
Perhaps you could include an architecture diagram to show how these components interact with each other at a higher level.